core/ordersync_subprotocols.go

```diff
@@ -0,0 +1,129 @@
+// which is expected to be an empty string on the first request.
+type FilteredPaginationRequestMetadata struct {
+	Page       int    `json:"page"`
+	SnapshotID string `json:"snapshotID"`
+}
+
+// FilteredPaginationResponseMetadata is the response metadata for the
+// FilteredPaginationSubProtocol. It keeps track of the current page and SnapshotID.
+type FilteredPaginationResponseMetadata struct {
+	Page       int    `json:"page"`
+	SnapshotID string `json:"snapshotID"`
+}
+
+// Name returns the name of the FilteredPaginationSubProtocol
+func (p *FilteredPaginationSubProtocol) Name() string {
+	return "/pagination-with-filter/version/0"
+}
+
+// HandleOrderSyncRequest returns the orders for one page, based on the page number
+// and snapshotID corresponding to the given request. This is
+// the implementation for the "provider" side of the subprotocol.
+func (p *FilteredPaginationSubProtocol) HandleOrderSyncRequest(ctx context.Context, req *ordersync.Request) (*ordersync.Response, error) {
+	var metadata *FilteredPaginationRequestMetadata
+	if req.Metadata == nil {
+		// Default metadata for the first request.
+		metadata = &FilteredPaginationRequestMetadata{
+			Page:       0,
+			SnapshotID: "",
+		}
+	} else {
+		var ok bool
+		metadata, ok = req.Metadata.(*FilteredPaginationRequestMetadata)
+		if !ok {
+			return nil, fmt.Errorf("FilteredPaginationSubProtocol received request with wrong metadata type (got %T)", req.Metadata)
+		}
+	}
+
+	ordersResp, err := p.app.GetOrders(metadata.Page, p.perPage, metadata.SnapshotID)
+	if err != nil {
+		return nil, err
+	}
+	orders := make([]*zeroex.SignedOrder, len(ordersResp.OrdersInfos))
+	for i, orderInfo := range ordersResp.OrdersInfos {
+		orders[i] = orderInfo.SignedOrder
+	}
+	// TODO(albrow): Filter orders
+	return &ordersync.Response{
+		Orders:   orders,
+		Complete: len(orders) == 0,
+		Metadata: &FilteredPaginationResponseMetadata{
+			Page:       metadata.Page,
+			SnapshotID: ordersResp.SnapshotID,
+		},
+	}, nil
+}
+
+// HandleOrderSyncResponse handles the orders for one page by validating them, storing them
+// in the database, and firing the appropriate events. It also returns the next request to
+// be sent. This is the implementation for the "requester" side of the subprotocol.
+func (p *FilteredPaginationSubProtocol) HandleOrderSyncResponse(ctx context.Context, res *ordersync.Response) (*ordersync.Request, error) {
+	if res.Metadata == nil {
+		return nil, errors.New("FilteredPaginationSubProtocol received response with nil metadata")
+	}
+	metadata, ok := res.Metadata.(*FilteredPaginationResponseMetadata)
+	if !ok {
+		return nil, fmt.Errorf("FilteredPaginationSubProtocol received response with wrong metadata type (got %T)", res.Metadata)
+	}
+	// TODO(albrow): Check that this order matches our current filter/topic
+	_, err := p.app.orderWatcher.ValidateAndStoreValidOrders(ctx, res.Orders, false, p.app.chainID)
```

Comment:
@albrow in order for our order propogation dashboard to continue to make sense, we need to emit similar logs here for orders discovered and stored via `ordersync`. Relevant code-snippet from `message_handler.go`:

```
validationResults, err := app.orderWatcher.ValidateAndStoreValidOrders(ctx, orders, false, app.chainID)
if err != nil {
	return err
}

for _, acceptedOrderInfo := range validationResults.Accepted {
	// If the order isn't new, we don't log it's receipt or adjust peer scores
	if !acceptedOrderInfo.IsNew {
		continue
	}
	msg := orderHashToMessage[acceptedOrderInfo.OrderHash]
	// If we've reached this point, the message is valid, we were able to
	// decode it into an order and check that this order is valid. Update
	// peer scores accordingly.
	log.WithFields(map[string]interface{}{
		"orderHash": acceptedOrderInfo.OrderHash.Hex(),
		"from":      msg.From.String(),
	}).Info("received new valid order from peer")
	log.WithFields(map[string]interface{}{
		"order":     acceptedOrderInfo.SignedOrder,
		"orderHash": acceptedOrderInfo.OrderHash.Hex(),
		"from":      msg.From.String(),
	}).Trace("all fields for new valid order received from peer")
}
```

Reply:
@fabioberger thanks for the tip. I just added a similar log message in this PR.